### PR TITLE
feat: show organization name for groups on user profile

### LINKED ISSUE #14213: Account settings page currently lists the current groups a user is in. Should either replace, or include which organizations
Multi-organizations make this listing of groups not a full accounting of a user's memberships on the product. We should probably list all the organizations and their groups within each organization.

Screenshot from `/settings/account` page

**Requirements**

- [ ] add additional detail to give org-scope without scrapping original design (keep design scope small)
- [ ] add org name and avatar to existing query
- [ ] Ensure storybook exists and update it

Screenshot from 2024-08-07 16-32-28

Related: https://github.com/coder/coder/pull/14200

### PR DESCRIPTION
**Pull request by aslilac**
Closes #14213

Show the organization the group is from on the user profile page for multi-org deployments

### CODE REVIEW

**Review by Emyrk — COMMENTED** [review]
Use `sqlc.embed`, and another idea is to rope `GetGroupByID` and `GetGroupsByOrgAndName` into the `GetGroups` query.

It feels weird, but it guarantees 1 `fetch` query on the backend, and therefore 1 group `type`. 

I did this with `OrganizationMembers` and use a wrapper func to convert the `:Many` to `:One` query:
https://github.com/coder/coder/blob/e000bd1d4a294502bdf8f0fd88ad2f506543db8e/coderd/httpmw/organizationparam.go#L123-L126

The alternative is making a view, but those are kinda a pain to maintain with migrations. I'll let you make that call based on where the types are used. It is just annoying to have `database.GetGroupRow` and `database.Group` types in the BE sometimes.

**Comment by aslilac** [comment]
> another idea is to rope GetGroupByID and GetGroupsByOrgAndName into the GetGroups query.

I am not against this, but it sounds like it might be a little out of my comfort zone, and probably big enough for a PR of its own

**Comment by aslilac** [comment]
> It is just annoying to have database.GetGroupRow and database.Group types in the BE sometimes.

Definitely agree. This is one of the reasons row-typing can be really nice on the backend, but we're not replacing Go anytime soon. :p

File: coderd/database/queries/groups.sql [hunk 1]
```diff
@@ -22,32 +22,36 @@ LIMIT
...
+								FROM
+										-- this view handles the 'everyone' group in orgs.
+										group_members_expanded
+								WHERE
+										group_members_expanded.group_id = groups.id
+								AND
+										group_members_expanded.user_id = @has_member_id
+						)
+				ELSE true
+		END
```
Comment (Emyrk) [inline coderd/database/queries/groups.sql:54]: Spacing seems to have gone a bit wack

File: coderd/database/queries/groups.sql [hunk 2]
```diff
@@ -22,32 +22,36 @@ LIMIT
...
 SELECT
-    *
+		sqlc.embed(groups),
+		organizations.name AS organization_name,
+		organizations.display_name AS organization_display_name
 FROM
-    groups
+		groups
+INNER JOIN
+		organizations ON ((groups.organization_id = organizations.id))
```
Comment (Emyrk) [inline coderd/database/queries/groups.sql:31]:
```suggestion
		organizations ON groups.organization_id = organizations.id
```

File: coderd/database/queries/groups.sql [hunk 3]
```diff
@@ -70,15 +74,15 @@ INSERT INTO groups (
...
+						gen_random_uuid(),
+						group_name,
+						@organization_id,
+						@source
 FROM
-    	    	UNNEST(@group_names :: text[]) AS group_name
+						UNNEST(@group_names :: text[]) AS group_name
 -- If the name conflicts, do nothing.
 ON CONFLICT DO NOTHING
 RETURNING *;
```
Comment (Emyrk) [inline coderd/database/queries/groups.sql:88]: Tabs vs spaces is so annoying in our sql files. We are inconsistent I know :cry:

File: enterprise/coderd/groups.go [hunk 4]
```diff
@@ -408,7 +408,7 @@ func (api *API) groupsByOrganization(rw http.ResponseWriter, r *http.Request) {
 // @Tags Enterprise
 // @Param organization query string true "Organization ID or name"
 // @Param has_member query string true "User ID or name"
-// @Success 200 {array} codersdk.Group
+// @Success 200 {array} codersdk.GroupWithOrganizationInfo
```
Comment (Emyrk) [inline enterprise/coderd/groups.go:411]: also fix this on the api above, `groupsByOrganization`

File: site/src/api/api.ts [hunk 5]
```diff
@@ -1603,14 +1603,20 @@ class ApiMethods {
...
 	/**
 	 * @param organization Can be the organization's ID or name
 	 */
-	getGroups = async (organization: string): Promise<TypesGen.Group[]> => {
+	getGroupsByOrganization = async (
+		organization: string,
+	): Promise<TypesGen.Group[]> => {
 		const response = await this.axios.get(
-			`/api/v2/organizations/${organization}/groups`,
+			`/api/v2/organization/${organization}/groups`,
```
Comment (Emyrk) [inline site/src/api/api.ts:1618]: Is this correct? It is `organizations` on the Go router.

https://github.com/coder/coder/blob/360d1a10b54242194a675e85607227f4e9f16b52/enterprise/coderd/coderd.go#L306-L306

File: coderd/database/queries/groups.sql (hunk 1 above)
Comment (aslilac) [reply coderd/database/queries/groups.sql:54]: yeah, this was one of the few bits of the file using spaces, so I just converted the whole thing to tabs

File: site/src/api/api.ts (hunk 5 above)
Comment (aslilac) [reply site/src/api/api.ts]: I just noticed this too 😅

File: enterprise/coderd/groups.go (hunk 4 above)
Comment (aslilac) [reply enterprise/coderd/groups.go:411]: good catch!

**Review by Emyrk — APPROVED** [review]

**Pull request merged**